app/models/concerns/terrier/embedder.rb

```diff
@@ -17,14 +17,52 @@ def embedded_fields
       @_embedded_fields
     end
 
-    def embeds_many(model_name_plural, options={})
-      model_name = model_name_plural.to_s.singularize
-      model = (options[:class_name] || model_name.classify).constantize
-      field_name = model_name_plural.to_s
-
-      embedded_fields[model_name_plural] = {
-          name: model_name_plural,
-          type: model
+    def embeds_one(model_name, options={})
+
+      model = (options[:class_name] || model_name.to_s.classify).constantize
+      field_name = model_name.to_s
+
+      embedded_fields[model_name] = {
+        name: model_name,
+        type: model
+      }
+
+      define_method field_name do
+        val = super()
+        if val.is_a? String
+          val = val.length>0 ? JSON.parse(val) : nil
+        end
+        if val.blank?
+          nil # may want to have this return a new, but empty instance of the embeded model?
+        elsif val.is_a? String
+          model.from_string(val)
+        else
+          model.from_attributes(val)
+        end
+      end
+
+      define_method "#{field_name}=" do |val|
+        if val.present?
+          if val.class.name.index('Hash')
```

Comment:
Is there a specific reason to do this string match? I would suspect just doing a `val.is_a? Hash` would be sufficient but maybe there's a specific edge case you're addressing here? If there is a specific case you're addressing, probably best to leave a comment saying why.

Reply:
Copied that logic from the `embeds_many` definition. Presumably, `#is_a? Hash` would be fine, but I think its in case its cast to any class containing 'Hash'?

Reply:
Hmm, fair enough. Just leave it how it is then since we don't know the exact reason why.